biojava-structure/src/main/java/org/biojava/nbio/structure/io/mmcif/DownloadChemCompProvider.java

```diff
@@ -166,62 +166,57 @@ private void split() throws IOException {
 		int counter = 0;
 		InputStreamProvider prov = new InputStreamProvider();
 
-		InputStream inStream = prov.getInputStream(f);
-
-		BufferedReader buf = new BufferedReader (new InputStreamReader (inStream));
+		try( BufferedReader buf = new BufferedReader (new InputStreamReader (prov.getInputStream(f)));
+				) {
+			String line = null;
+			line = buf.readLine ();
+			StringWriter writer = new StringWriter();
```

Comment:
Yep! I'm going to milk the 1 month of Java 7 development for all its worth. I'm also using [type inference](https://docs.oracle.com/javase/tutorial/java/generics/genTypeInference.html) in all my code from now on.